test/e2e/operator_test.go

```diff
@@ -1907,8 +1907,8 @@ func TestHTTPHeaderCapture(t *testing.T) {
 	if err := kclient.List(context.TODO(), podList, client.MatchingLabelsSelector{Selector: selector}); err != nil {
 		t.Fatalf("failed to list pods for ingresscontroller: %v", err)
 	}
-	if len(podList.Items) < 1 {
-		t.Fatalf("found no pods for ingresscontroller: %v", err)
+	if len(podList.Items) != 1 {
+		t.Fatalf("expected ingress controller %s to have exactly 1 router pod, but it has %d", ic.Name, len(podList.Items))
```

Comment:
It might be helpful to log the pod names as well.  Then someone who observed the test failure could refer to the events from must-gather to investigate the specific pods.  
```suggestion
		var podNames []string
		for i := range podList.Items {
			podNames = append(podNames, podList.Items[i].Name)
		}
		t.Fatalf("expected ingress controller %s to have exactly 1 router pod, but it has %d: %s", ic.Name, len(podList.Items), strings.Join(podNames, ", "))
```